ui/raidboss/emulator/ui/EmulatedPartyInfo.js

```diff
@@ -286,19 +347,54 @@ export default class EmulatedPartyInfo extends EventBus {
         false);
   }
 
-  _wrapCollapse(label, $obj, onclick) {
+  _wrapCollapse(params = {
+    time: undefined,
+    name: undefined,
+    icon: undefined,
+    text: undefined,
+    classes: undefined,
+    $obj,
+    onclick: undefined,
+  }) {
```

Comment:
```suggestion
  _wrapCollapse(params = {
    // all parameters are optional except for $obj
    time: undefined,
    name: undefined,
    icon: undefined,
    text: undefined,
    classes: undefined,
    $obj: undefined, // required
    onclick: undefined,
  }) {
```

I think this is just documentation and the values and keys here are not used in practice.  So, maybe the answer is just more documentation?  I'm not sure I understood the nuance that $obj was required by initializing it with shorthand until you explained it to me.

Reply:
Hmm, given that it's the only 100% required parameter, maybe it should be a separate function argument, e.g.:
```js
  _wrapCollapse($obj, params = {
    // All params properties are optional
    time: undefined,
    name: undefined,
    icon: undefined,
    text: undefined,
    classes: undefined,
    onclick: undefined,
  }) {
```

Reply:
Whichever of these two options makes more sense to you, I'll switch to.

Reply:
I think I'd say just put it all in params with a comment, but I only weakly prefer that.  As you say, with typescript this is all kind of moot, and so mostly I just want something that's a little bit more readable and will remind me (or you) in the future of what these parameters mean when it becomes typescript.

Reply:
How's this?
![image](https://user-images.githubusercontent.com/6119598/113522613-07858a00-9570-11eb-8af1-78fa98b8b706.png)

Reply:
Even better, but I still would prefer it to say `${obj}: undefined` for shorthand consistency.